src/vm/class.cpp

```diff
@@ -1172,6 +1172,58 @@ bool MethodTable::IsHFA()
 }
 #endif // !FEATURE_HFA
 
+//*******************************************************************************
+int MethodTable::GetVectorSize()
+{
+    // This is supported for finding HVA types for Arm64. In order to support the altjit,
+    // we support this on 64-bit platforms (i.e. Arm64 and X64).
+#ifdef _TARGET_64BIT_
+    if (IsIntrinsicType())
+    {
+        LPCUTF8 namespaceName;
+        LPCUTF8 className = GetFullyQualifiedNameInfo(&namespaceName);
+        int vectorSize = 0;
+
+        if (strcmp(className, "Vector`1") == 0)
```

Comment:
What about `Vector2`, `Vector3`, and `Vector4`? Are they staying as homogeneous aggregates or will they also be treated as `short vector` for the purpose of passing?

Reply:
They are explicitly structs of floats, so they will stay has HFAs of float. (That's how I read the ABI spec).

Reply:
I see your comment about that these are staying as HFAs today.

The upside is that this means it can't be a breaking change; but the downside is that we can't make the existing System.Numerics code less expensive...
* Most of them expose public fields; and something like `Matrix4x4` could have potentially been an HVA, but it will continue being passed `byref`.
* Maybe this can just be resolved with the generic `Vector2<T>`, `Vector3<T>`, and `Vector4<T>` proposal...

Reply:
However, unlike the other types, these are existing interop types (and were designed to be), so they need to continue to follow the ARM64 ABI.